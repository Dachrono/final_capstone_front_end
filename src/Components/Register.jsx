import { useDispatch } from 'react-redux';
import { Link, useNavigate } from 'react-router-dom';
import { createUser } from '../Redux/Slices/usersSlice';

const user = (userName, nickname) => ({
  name: userName,
  username: nickname,
});

const Register = () => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const CreateUser = (e) => {
    const btn = e.target;
    const dataForm = btn.parentElement.parentElement;
    const name = dataForm.querySelector('#name').value;
    const username = dataForm.querySelector('#username').value;
    const msg = dataForm.querySelector('#message-error');
    const response = dispatch(createUser(user(name, username)));

    response.then((data) => {
      if (typeof data.payload !== 'object') {
        msg.textContent = 'User already exist';
      } else {
        msg.textContent = '';
        navigate('/MainPage');
      }
    });
  };

  return (
    <section className="image-back d-flex-col">
      <div className="d-flex-col form-cont gap">
        <h2 className="title-form">Register</h2>
        <form className="d-flex-col gap" onSubmit={(ev) => ev.preventDefault()}>
          <span id="message-error" />
          <input id="name" type="text" placeholder="Name" className="form-input" />
          <input id="username" type="text" placeholder="Username" className="form-input" />
          <div className="btn-cont gap">
            <button type="submit" className="form-submit" onClick={(e) => CreateUser(e)}>Register</button>
            <Link to="/" className="form-submit d-flex-col">Log In</Link>
          </div>
        </form>
      </div>
    </section>
  );
};

export default Register;
