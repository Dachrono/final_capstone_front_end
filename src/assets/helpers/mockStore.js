import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';

export const experiencesData = [
  {
    experienceName: 'Experience 1',
    experienceDetails: 'Super test Experience',
    description: 'Best Experience',
    experienceFee: 15.50,
    addTestingFee: 20,
    reserveFullTable: 50,
    guests: 2,
    experienceImage: 'https://i.postimg.cc/bNX1CZRC/casa-coeli.jpg',
  },
  {
    experienceName: 'Experience 2',
    experienceDetails: 'Super 2 test Experience 2',
    description: 'Second Best Experience',
    experienceFee: 35.50,
    addTestingFee: 10,
    reserveFullTable: 40,
    guests: 20,
    experienceImage: 'https://i.postimg.cc/bNX1CZRC/casa-coeli.jpg',
  },
];

export const reservationsData = [
  {
    userID: 1,
    experienceID: 1,
    reserveDate: 2023-12-12,
    reserveCity: 'Cancun',
  },
  {
    userID: 1,
    experienceID: 2,
    reserveDate: 2023-12-25,
    reserveCity: 'Caracas',
  },
];

export const usersData = [
  {
    userID: 1,
    name: 'Test name1',
    username: 'Test username 1',
  },
];

export const experiencesFetchMockData = [
  {
    experienceName: 'Experience 1',
    experienceDetails: 'Super test Experience',
    description: 'Best Experience',
    experienceFee: 15.50,
    addTestingFee: 20,
    reserveFullTable: 50,
    guests: 2,
    experienceImage: 'https://i.postimg.cc/bNX1CZRC/casa-coeli.jpg',
  },
  {
    experienceName: 'Experience 2',
    experienceDetails: 'Super 2 test Experience 2',
    description: 'Second Best Experience',
    experienceFee: 35.50,
    addTestingFee: 10,
    reserveFullTable: 40,
    guests: 20,
    experienceImage: 'https://i.postimg.cc/bNX1CZRC/casa-coeli.jpg',
  },  
];

export const reservationsFetchMockData = [
  {
    reserveDate: 2023-12-12,
    reserveCity: 'Cancun',
  },
  {
    reserveDate: 2023-12-25,
    reserveCity: 'Caracas',
  },  
];

export const usersFetchMockData = [
  {
    name: 'Test name1',
    username: 'Test username 1',
  },
];

const mockStore = configureStore([]);
const store = mockStore({
  experiences: {
    error: null,
    status: 'idle',
    experiences: experiencesData,
  },
  reservations: {
    loading: false,
    error: null,
    currentReservations: reservationsData,
  },
  users: {
    user: usersData,
    loading: null,
    status: 'idle',
    error: null,
  }
});

export const usersMockData = {
  user: usersData,
  loading: null,
  status: 'idle',
  error: null,
};

export const experienceMockData = {
  experiences: experiencesData,
  loading: false,
  error: null,
};

export const reservationsMockData = {
  reservations: reservationsData,
  loading: false,
  error: null,
};

export default store;

export const initialStore = mockStore({
  experiences: { 
    experiences: [], 
    status: 'idle', 
    error: null 
  },
  reservations: {
    currentReservations: [], 
    loading: false, 
    error: null 
  },
  users: {
    user: { id: null, name: null, username: null },
    loading: null,
    status: 'idle',
    error: null
  }
});