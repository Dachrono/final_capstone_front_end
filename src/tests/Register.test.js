import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { initialStore } from '../assets/helpers/mockStore';
import Register from '../Components/Register';

describe('The Register component', () => {
  test('should render correctly into the DOM', async () => {
    const { registerComponent } = render(
      <MemoryRouter>
        <Provider store={initialStore}>
          <Register />
        </Provider>
      </MemoryRouter>,
    );
    expect(registerComponent).toMatchSnapshot();
  });
});
