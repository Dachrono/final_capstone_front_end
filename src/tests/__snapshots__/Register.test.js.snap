// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Register component should render correctly into the DOM 1`] = `undefined`;
